### PR TITLE
Add deprecation comment to PersistentVolumeReclaimRecycle

### LINKED ISSUE #59060: The `Recycle` persistentVolumeReclaimPolicy should have deprecation notice.
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:
The `persistentVolumeReclaimPolicy` called `Recycle` is being deprecated. See [announcement here](https://groups.google.com/forum/#!topic/kubernetes-dev/uexugCza84I). The documentation (comments) in the API (https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L478) should mention this.

**What you expected to happen**:
API documentation should indicate `Recycle` policy is being deprecated.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by ayushpateria**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #59060

**Special notes for your reviewer**:

**Release note**:

```release-note
Announcing the deprecation of the recycling reclaim policy.
```

### CODE REVIEW

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -475,6 +475,7 @@ type PersistentVolumeReclaimPolicy string
 const (
 	// PersistentVolumeReclaimRecycle means the volume will be recycled back into the pool of unbound persistent volumes on release from its claim.
 	// The volume plugin must support Recycling.
+	// This is deprecated.
```
Comment (linyouchong) [inline pkg/apis/core/types.go]: The description is too simple, my suggestion:
'DEPRECATED: The PersistentVolumeReclaimRecycle called Recycle is being deprecated. See announcement here: https://groups.google.com/forum/#!topic/kubernetes-dev/uexugCza84I`

Comment (ayushpateria) [reply pkg/apis/core/types.go]: Thanks! I pushed a commit with the changes you mentioned.

**Comment by ayushpateria** [comment]
@mlmhl Added! Please check.

**Comment by saad-ali** [comment]
For API approval:

/assign @thockin

**Pull request merged**